rust/src/user/user_manager.rs

```diff
@@ -81,119 +49,107 @@ impl UserManager {
+    /// Retrieve a user with the given name.
     ///
     /// # Arguments
     ///
-    /// * `username` -- The name of the user to be created
-    /// * `password` -- The password of the user to be created
+    /// * `username` -- The name of the user to retrieve
     ///
     /// # Examples
     ///
     /// ```rust
-    /// driver.users.create(username, password).await;
+    /// driver.users.get(username).await;
     /// ```
     #[cfg_attr(feature = "sync", maybe_async::must_be_sync)]
-    pub async fn create(&self, username: impl Into<String>, password: impl Into<String>) -> Result {
-        let username = username.into();
-        let password = password.into();
-        self.run_any_node(|server_connection: ServerConnection| {
-            let username = username.clone();
-            let password = password.clone();
-            async move { server_connection.create_user(username, password).await }
-        })
-        .await
+    pub async fn get(&self, username: impl Into<String>) -> Result<Option<User>> {
+        let uname = username.into();
+        let mut error_buffer = Vec::with_capacity(self.server_connections.len());
+        for (server_id, server_connection) in self.server_connections.iter() {
+            match server_connection.get_user(uname.clone()).await {
+                Ok(res) => return Ok(res),
+                Err(err) => error_buffer.push(format!("- {}: {}", server_id, err))
+            }
+        }
+        Err(ConnectionError::ServerConnectionFailedWithError { error: error_buffer.join("\n") })?
     }
 
-    /// Deletes a user with the given name.
-    ///
-    /// # Arguments
-    ///
-    /// * `username` -- The name of the user to be deleted
+    /// Retrieves all users which exist on the TypeDB server.
     ///
     /// # Examples
     ///
     /// ```rust
-    /// driver.users.delete(username).await;
+    /// driver.users.all().await;
     /// ```
     #[cfg_attr(feature = "sync", maybe_async::must_be_sync)]
-    pub async fn delete(&self, username: impl Into<String>) -> Result {
-        let username = username.into();
-        self.run_any_node(|server_connection: ServerConnection| {
-            let username = username.clone();
-            async move { server_connection.delete_user(username).await }
-        })
-        .await
+    pub async fn all(&self) -> Result<Vec<User>> {
+        let mut error_buffer = Vec::with_capacity(self.server_connections.len());
+        for (server_id, server_connection) in self.server_connections.iter() {
+            match server_connection.all_users().await {
+                Ok(res) => return Ok(res),
+                Err(err) => error_buffer.push(format!("- {}: {}", server_id, err))
+            }
+        }
+        Err(ConnectionError::ServerConnectionFailedWithError { error: error_buffer.join("\n") })?
     }
 
-    /// Retrieve a user with the given name.
+    /// Create a user with the given name &amp; password.
```

Comment:
Is `&amp;` somehow parsed in the result docs?

Reply:
It was not written by me, but I'd assume so since we have another instance in the C driver. It must have been deliberate instead of a typo.